Unicode UI/MitSciUnicodeUI.pax

```diff
@@ -0,0 +1,1283 @@
+				[ShellView createHookBlock: nil].
+			^newView]
+		ifFalse: [resourceIdentifier := ResourceIdentifier class: self name: aString.
+			^resourceIdentifier loadWithContext: aParentView]! !
+!Presenter class categoriesFor: #isUnicodeMode!public! !
+!Presenter class categoriesFor: #loadViewResource:inContext:!public! !
+
+!Prompter class methodsFor!
+
+isUnicodeMode
+
+	"cdemers 11/30/2016 decide the unicode type based on the active window."
+	^View active isUnicodeView. ! !
+!Prompter class categoriesFor: #isUnicodeMode!public! !
+
+!SymbolStringSearchPolicy methodsFor!
+
+compare: operand1 with: operand2
+	"Answer whether the <Object>, operand1, is considered equivalent to the <Object> argument,
+	operand2, by this search policy."
+	"cdemers 11/23/2016 Revised for Unicode support.  This may be innefficient."
+
+	^(operand1 asString trueCompare: operand2 asString) == 0
+!
+
+hash: operand
+	"Answer a suitable hash value for the <Object>, operand, under this search policy."
+	"cdemers 11/23/2016 Revised to accomodate UnicodeStrings"
+
+	^operand hash! !
+!SymbolStringSearchPolicy categoriesFor: #compare:with:!comparing!public! !
+!SymbolStringSearchPolicy categoriesFor: #hash:!comparing!public! !
+
+!TabView methodsFor!
+
+getItem: aninteger 
+	| tcItem |
+
+	"cdemers 12/5/2016 Revised for Unicode support."
+	tcItem := TCITEM new.
+	tcItem
+		text: (self stringClass newFixed: 256);
+		maskIn: tcItem imageValidMask.
+	self 
+		sendMessage: (self isUnicodeView ifTrue: [TCM_GETITEM-5+60] ifFalse: [TCM_GETITEM  ])
+		wParam: aninteger - 1
+		lpParam: tcItem asParameter.
+	^tcItem!
+
+tcmInsertItem: tcItem atOffset: anInteger
+	"Private - Insert a tab defined by tcItem to the receiver at offset anInteger."
+
+	"cdemers 12/5/2016 Revised for Unicode support."
+
+	^self sendMessage: ((tcItem text isKindOf: UnicodeString) ifTrue: [TCM_INSERTITEM -7+62] ifFalse: [TCM_INSERTITEM]) 
+		wParam: anInteger lpParam: tcItem asParameter!
+
+updateItem: anObject atIndex: aninteger 
+	"Re-render the specified item, which is at the specified <integer> index in the list."
+
+	"cdemers 12/5/2016 Revised for Unicdode."
+
+	| tcItem |
+	tcItem := self makeTcItemStruct: anObject.
+	self 
+		sendMessage: ((tcItem text isKindOf: UnicodeString) ifTrue: [TCM_SETITEM -6+61] ifFalse: [TCM_SETITEM])
+		wParam: aninteger - 1
+		lpParam: tcItem asParameter! !
+!TabView categoriesFor: #getItem:!helpers!private! !
+!TabView categoriesFor: #tcmInsertItem:atOffset:!adding!private! !
+!TabView categoriesFor: #updateItem:atIndex:!event handling!public! !
+
+!UnicodeCharacter methodsFor!
+
+isSeparator
+	"cdemers 11/30/2016 Taken and revised from Character.  Not sure if there may be a broader meaning in Unicode."
+	"Answer whether the receiver is a separator character (i.e. whitespace)."
+
+	^self codePoint == 32 or: [self codePoint >= 9 and: [self codePoint <= 13]]! !
+!UnicodeCharacter categoriesFor: #isSeparator!public! !
+
+!UnicodeString methodsFor!
```

Comment:
It would be worth pulling out the additions of `_collate:`, `trueCompare:` and `includes:` and PRing those into the base system, since they are just missing and should be added regardless.